src/router.rs

```diff
@@ -131,15 +136,27 @@ impl Router {
     /// Route will match any method, including gibberish.
     /// In case of ambiguity, handlers specific to methods will be preferred.
     pub fn any<S: AsRef<str>, H: Handler, I: AsRef<str>>(&mut self, glob: S, handler: H, route_id: I) -> &mut Router {
-        self.mut_inner().wildcard.add(glob.as_ref(), Box::new(handler));
+        self.mut_inner().wildcard.add(glob.as_ref(), Arc::new(handler));
         self.route_id(route_id.as_ref(), glob.as_ref());
         self
     }
 
     fn recognize(&self, method: &method::Method, path: &str)
-                     -> Option<Match<&Box<Handler>>> {
-        self.inner.routers.get(method).and_then(|router| router.recognize(path).ok())
-            .or(self.inner.wildcard.recognize(path).ok())
+                      -> Result<Match<&Arc<Handler>>, IronError> {
```

Comment:
In `handle` i like to change `Err(IronError::new(NoRoute, status::NotFound))` to `Err(IronError::new(RouterError::NotFound, status::NotFound))` this more clearly. But unfortunately, many developers can already use the structure `NoRoute` like in there https://github.com/iron/router/blob/master/examples/custom_404.rs#L19 . I also like to change ` Some(IronError::new(TrailingSlash, (status::MovedPermanently, Redirect(url))))` to 
`Some(IronError::new(RouterError::TrailingSlash, (status::MovedPermanently, Redirect(url))))` . Perhaps we can mark these structures as deprecated and remove them in the future.